d2common/d2math/d2vector/position.go

```diff
@@ -37,36 +45,47 @@ func (p *Position) checkValues() {
 }
 
 // World is the position, where 1 = one map tile.
+// unused
 func (p *Position) World() *Vector {
 	return &p.Vector
 }
 
-// Tile is the tile position, always a whole number.
+// Tile is the tile position, always a whole number. (tileX, tileY)
 func (p *Position) Tile() *Vector {
 	c := p.World().Clone()
 	return c.Floor()
 }
 
 // TileOffset is the offset from the tile position, always < 1.
+// unused
 func (p *Position) TileOffset() *Vector {
 	c := p.World().Clone()
 	return c.Subtract(p.Tile())
 }
 
-// SubWorld is the position, where 5 = one map tile.
+// SubWorld is the position, where 5 = one map tile. (locationX, locationY)
 func (p *Position) SubWorld() *Vector {
 	c := p.World().Clone()
 	return c.Scale(subTilesPerTile)
 }
 
 // SubTile is the tile position in sub tiles, always a multiple of 5.
+// unused
 func (p *Position) SubTile() *Vector {
-	c := p.Tile().Clone()
-	return c.Scale(subTilesPerTile)
+	return p.Tile().Scale(subTilesPerTile)
 }
 
 // SubTileOffset is the offset from the sub tile position in sub tiles, always < 1.
+// unused
 func (p *Position) SubTileOffset() *Vector {
-	c := p.SubWorld().Clone()
-	return c.Subtract(p.SubTile())
+	return p.SubWorld().Subtract(p.SubTile())
+}
+
+// TODO: understand this and maybe improve/remove it
+// SubTileOffset() + 1. It's used for rendering. It seems to always do this:
+// 	v.offsetX+int((v.subcellX-v.subcellY)*16),
+//	v.offsetY+int(((v.subcellX+v.subcellY)*8)-5),
+// ^ Maybe similar to Viewport.OrthoToWorld? (subCellX, subCellY)
```

Comment:
@essial  I'm interested in whether this is correct and learning more about mapEntity.subCellX/Y and the position conversion functions in viewport.go (*ToOrtho in particular). I commented those in another PR but unsure if my comments are correct,